src/dlls/mscoree/unixinterface.cpp

```diff
@@ -127,6 +130,18 @@ static void ConvertConfigPropertiesToUnicode(
 
     for (int propertyIndex = 0; propertyIndex < propertyCount; ++propertyIndex)
     {
+        if (strcmp(propertyKeys[propertyIndex], "BUNDLE_PROBE") == 0)
```

Comment:
I don't like how this makes the function do something it is not supposed to do just to save an iteration over the few properties. This function should be what its name says - just convert property keys and values to unicode. 
I would much more prefer passing the bundle probe address formatted as hex string and parse it from the string in the coreclr_initialize. This is not perf sensitive code in any way, so we should avoid hacks like passing a non-string in an array of strings.

Reply:
I certainly don't like this way of passing in the bundle-probe (or the implementation of this function). This is pretty much a hack for now -- that communicates the bundle-probe address without changing the host layer interfaces.

In the product, I think we should design (an additional) host-interface, to pass in a structure of well-typed members, instead of a list of strings. We should design the host-interface to accommodate this and other upcoming features.  Until then, my opinion is to let the hack remain, instead of polishing something that'll go away.

Reply:
Ok, if this is just a temporary way, then let's keep it for now. Can you please create an issue for fixing that so that we don't forget about the need to fix it?

Reply:
> we should design (an additional) host-interface

For new host-interface designs, we should also take into account preloading scenario described in https://github.com/dotnet/coreclr/pull/25725

cc @alpencolt @kbaladurin

Reply:
I created an issue to track this: https://github.com/dotnet/coreclr/issues/26752